src/program.cpp

```diff
@@ -415,6 +415,54 @@ bool spir_binary::load_descriptor_map(
             str << entry;
             cvk_debug("DMAP line: %s", str.str().c_str());
         }
+
+        if (entry.kind == clspv::version0::DescriptorMapEntry::Kind::Sampler) {
+            sampler_desc desc;
+            desc.descriptorSet = entry.descriptor_set;
+            desc.binding = entry.binding;
+
+            desc.normalized_coords = entry.sampler_data.mask & 0x1;
+
+            switch (entry.sampler_data.mask & 0xe) {
+            case clspv::version0::CLK_ADDRESS_NONE:
+                desc.addressing_mode = CL_ADDRESS_NONE;
+                break;
+            case clspv::version0::CLK_ADDRESS_CLAMP_TO_EDGE:
+                desc.addressing_mode = CL_ADDRESS_CLAMP_TO_EDGE;
+                break;
+            case clspv::version0::CLK_ADDRESS_CLAMP:
+                desc.addressing_mode = CL_ADDRESS_CLAMP;
+                break;
+            case clspv::version0::CLK_ADDRESS_MIRRORED_REPEAT:
+                desc.addressing_mode = CL_ADDRESS_MIRRORED_REPEAT;
+                break;
+            case clspv::version0::CLK_ADDRESS_REPEAT:
+                desc.addressing_mode = CL_ADDRESS_REPEAT;
+                break;
+            default:
+                cvk_error("Invalid sampler addressing mode: %d",
+                          entry.sampler_data.mask & 0xe);
```

Comment:
You could have introduced a local variable to avoid repeating the mask and magic number.